processor/pom.xml

```diff
@@ -375,9 +375,10 @@
                 <dependency>
                     <groupId>javax.xml.bind</groupId>
                     <artifactId>jaxb-api</artifactId>
-                    <version>2.3.1</version>
-                    <scope>provided</scope>
-                    <optional>true</optional>
```

Comment:
> We do not want to have that

Yes, I'm aware of that.

> Removing this will make MapStruct compile depend on this dependency.

Well, it shouldn't have. I thought that by moving the `scope` and the `optional` tags to the parent `pom.xml` the individual dependencies will inherit both the `scope` and the `optional` tags. At least I knew that that was the case for the `scope` tag. But unfortunately, I didn't double-check if that was also the case for the `optional` tag.

Thanks to this comment of yours, I dug deeper into the way Maven handles the inheritance of the `optional` tag. And I have found that the `optional` tag doesn't get inherited from the `dependencyManagement` at all (can be verified by running `./mvnw help:effective-pom -Dverbose`). It is just ignored. As far as I know, this behavior is not documented (because I assume it should not work like that by design). There's also a 10 year old opened issue that should solve this - https://issues.apache.org/jira/browse/MNG-5227. But it keeps being put off.

Given the above, I moved both tags back to the individual dependencies.

Reply:
Thanks @ibogdanchikov. I wasn't aware about the details of `optional`. In any case, I prefer having the `scope` and `optional` in the lowest possible place, otherwise it makes it complicated to understand the values